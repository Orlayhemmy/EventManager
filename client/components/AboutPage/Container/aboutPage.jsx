import React from 'react';
import Navbar from '../../Navbar/Index.jsx';
import Content from '../Template/aboutPageContent';
import Footer from '../../Footer/Index';

export default class AboutPage extends React.Component {
  render() {
    const { pathname } = this.props.location;
    return (
      <div id="about-page">
        <Navbar path={pathname} />
        <Content />
        <Footer />
      </div>
    );
  }
}
