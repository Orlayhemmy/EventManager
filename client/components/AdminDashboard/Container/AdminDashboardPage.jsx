import React from 'react';
import AdminContent from '../Template/Content/AdminContent';
import Navbar from '../../Navbar/Container/Navbar';
import Footer from '../../Footer/Footer';
/**
 * @memberof AdminDashboarPage
 * @method render
 */
const AdminDashboardPage = (props) => {
  const { location: { pathname } } = props;
  return (
    <div id="center-page">
      <Navbar path={pathname} />
      <AdminContent path={pathname} />
      <Footer />
    </div>
  );
};
export default AdminDashboardPage;
