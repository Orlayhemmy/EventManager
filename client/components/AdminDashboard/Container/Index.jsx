import React from 'react';
import AdminContent from '../Template/Content/AdminContent';
import Navbar from '../../Navbar/Container/navbar';
import Footer from '../../Footer/footer';
/**
 * @memberof AdminDashMethod
 * @method render
 */
export default class AdminDashMethod extends React.Component {
  render() {
    const { pathname } = this.props.location;
    return (
      <div id="center-page">
        <Navbar path={pathname} />
        <AdminContent pathname={pathname} />
        <Footer />
      </div>
    );
  }
}
