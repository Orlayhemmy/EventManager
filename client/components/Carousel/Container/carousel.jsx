import React from 'react';

const Carousel = (centerState) => {
  let centerList;
  let carouselIndicator;
  const { centers } = centerState;
  if (centers.length > 0) {
    centerList = centers.map((center, index) => {
      const { centerName, location, imageUrl } = center;
      return (
        <div
          className={index === 0 ? 'carousel-item active' : 'carousel-item'}
          key={index} // eslint-disable-line
        >
          <img className="d-block w-100 h-350" src={imageUrl} alt={centerName} />
          <div className="carousel-caption d-none d-md-block">
            <h5>
              {centerName}
            </h5>
            <p>
              {location}
            </p>
          </div>
        </div>
      );
    });
    carouselIndicator = centers.map((center, index) => (
      <li
        data-target="#carouselExampleIndicators"
        data-slide-to={index}
        className={index === 0 ? 'active' : ''}
        key={index} // eslint-disable-line
      />
    ));
  }
  return (
    <div className="col-lg-6 form">
      <h1>
        Our Centers
      </h1>
      <div
        id="carouselExampleIndicators"
        className="carousel slide"
        data-ride="carousel"
      >
        <ol className="carousel-indicators">
          {carouselIndicator}
        </ol>
        <div className="carousel-inner">
          {centerList}
        </div>
        <a
          className="carousel-control-prev"
          href="#carouselExampleIndicators"
          role="button"
          data-slide="prev"
        >
          <span className="carousel-control-prev-icon" aria-hidden="true" />
          <span className="sr-only">
          Previous
          </span>
        </a>
        <a
          className="carousel-control-next"
          href="#carouselExampleIndicators"
          role="button"
          data-slide="next"
        >
          <span className="carousel-control-next-icon" aria-hidden="true" />
          <span className="sr-only">
            Next
          </span>
        </a>
      </div>
    </div>
  );
};
export default Carousel;
