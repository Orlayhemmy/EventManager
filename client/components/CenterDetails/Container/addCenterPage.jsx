import React from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import swal from 'sweetalert2';
import CenterForm from '../Template/Form/addCenterForm';
import Navbar from '../../Navbar/Container/navbar';
import Footer from '../../Footer/footer';

/**
 * @description AddCenterPage component
 */
export class AddCenterPage extends React.Component {
  render() {
    const { user: { isAdmin }, isAuth } = this.props.userState;
    // Check if user is logged in and is also an Admin
    if (!isAuth) {
      return <Redirect to="/" />;
    } else if (!isAdmin) {
      return <Redirect to="/dashboard" />;
    }
    const { status, message } = this.props.center;
    const { pathname } = this.props.location;

    if (status === 201) {
      swal(message);
      return <Redirect to="/admin-centers" />;
    }
    return (
      <div id="add-center">
        <Navbar path={pathname} />
        <div class="container">
          <div class="row">
            <div className="col-lg-6">
              <div class="form-outer text-center">
                <div class="form-inner">
                  <span class="logo">
                    <strong class="text-primary">add a new center</strong>
                  </span>
                  <hr />
                  <CenterForm path={pathname} />
                </div>
              </div>
            </div>
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}

const propTypes = {
  userState: PropTypes.object.isRequired,
  center: PropTypes.object.isRequired
};
const mapStateToProps = state => ({
  userState: state.auth,
  center: state.center
});
AddCenterPage.propTypes = propTypes;

export default connect(
  mapStateToProps,
  {}
)(AddCenterPage);
