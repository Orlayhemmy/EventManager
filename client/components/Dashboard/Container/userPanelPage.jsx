/* eslint disable */
import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Redirect, Link } from 'react-router-dom';
import swal from 'sweetalert';
import { getEvents, setCurrentEvent } from '../../../actions/eventActions';
import EventForm from '../../EventPage/Template/Form/editEventForm';
import Navbar from '../../Navbar/Container/navbar';
import Footer from '../../Footer/footer';
import DeleteModal from '../../Modal/Container/deleteModal';
import { clearCenterStorage } from '../../../actions/centerActions';
import Modal from '../../Flash/modal';
import { logout } from '../../../actions/userActions';
import { getActivity } from '../../../actions/activityActions';
import DashboardContent from '../Template/Content/UserEvents';
import DashboardNotifications from '../Template/Content/UserNotifications';

/**
 * @description Dashboard component
 */
export class Dashboard extends React.Component {
  constructor() {
    super();
    this.state = {
      counter: 0
    };
    this.onClick = this.onClick.bind(this);
    this.onDelete = this.onDelete.bind(this);
    this.nextEvents = this.nextEvents.bind(this);
  }
  /**
   * @memberof Dashboard
   * @method componentWillMount
   * @description it calls an action
   * @param {void}
   * @returns {void}
   */
  componentWillMount() {
    this.props.clearCenterStorage();
    this.props.getEvents(0);
    this.props.getActivity(this.props.auth.user.id);
  }
  /**
   * @memberof Dashboard
   * @method onClick
   * @description it calls an action
   * @param {object} event
   * @returns {void}
   */
  onClick(e) {
    this.props.setCurrentEvent(e.target.id, e.target.parentNode.id);
  }
  /**
   * @memberof Dashboard
   * @method componentDidUpdate
   * @description it checks some conditions when component updates
   * @returns {void}
   */
  componentDidUpdate() {
    if (this.props.userEvent.status === 200) {
      $(document).ready(function() {
        $('#deleteModal').modal('hide');
      });
      swal(this.props.userEvent.message);
    }
  }
  /**
   * @memberof Dashboard
   * @method onDelete
   * @description it calls an action
   * @param {object} event
   * @returns {void}
   */
  onDelete(e) {
    const eventData = {
      eventId: e.target.id,
      eventName: e.target.parentNode.id
    };
    this.props.eventSelected(eventData);
  }
  /**
   * @memberof Dashboard
   * @method showDiv
   * @description it toggles div display
   * @param {object} event
   */
  showHiddenDiv(e) {
    const targetDiv = e.target.id;
    const div = document.getElementById(targetDiv);
    div.hidden = !div.hidden;
  }
  /**
   * @memberof Dashboard
   * @method logout
   * @description it calls a logout action
   * @param {object} event
   * @returns {void}
   */
  logout(e) {
    this.props.logout();
  }
  /**
   * @memberof Dashboard
   * @method nextEvents
   * @description it fetches the next centers
   * @returns {void}
   */
  nextEvents(e) {
    if (e.target.id === 'next') {
      this.setState({
        counter: this.state.counter + 1
      });
      this.props.getEvents(++this.state.counter);
    } else {
      this.setState({
        counter: this.state.counter - 1
      });
      this.props.getEvents(--this.state.counter);
    }
  }
  /**
   * @memberof Dashboard
   * @method render
   * @description it renders the component
   * @returns the HTML of Dashboard
   */
  render() {
    const { status } = this.props.userEvent;
    if (!this.props.auth.isAuth) {
      return <Redirect to="/" />;
    }
    if (status === 498 || status === 403) {
      this.logout();
    }
    const { pathname } = this.props.location;
    const { activities } = this.props.activity;
    const { message } = this.props.userEvent;
    return (
      <div id="dashboard">
        <Navbar />
        <div className="container">
          <div className="row pt-4">
            <div className="col-lg-9 col9-bg">
              <div className="row event">
                <h1>Dashboard</h1>
                <h3 className="main-color">List of events booked</h3>
                <hr />
              </div>
              <div className="row">
                <DashboardContent
                  userEvent={this.props.userEvent}
                  counter={this.state.counter}
                  onClick={this.onClick}
                  onDelete={this.onDelete}
                  nextEvents={this.nextEvents}
                />
                <DeleteModal path={pathname} />
                <Modal message={this.props.userEvent.message} />
              </div>
            </div>
            <div className="col-lg-3 col3-bg">
              <div className="p-4 fw">
                <h2>Notifications</h2>
              </div>
              <DashboardNotifications
                activities={activities}
                onClick={this.onClick}
              />
            </div>
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}
const propTypes = {
  auth: PropTypes.object.isRequired,
  userEvent: PropTypes.object.isRequired,
  activity: PropTypes.object.isRequired,
  getEvents: PropTypes.func.isRequired,
  clearCenterStorage: PropTypes.func.isRequired,
  setCurrentEvent: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
  getActivity: PropTypes.func.isRequired
};
const mapStateToProps = state => ({
  auth: state.auth,
  userEvent: state.event,
  activity: state.adminActivity
});
Dashboard.propTypes = propTypes;

export default connect(
  mapStateToProps,
  {
    setCurrentEvent,
    logout,
    getEvents,
    getActivity,
    clearCenterStorage
  }
)(Dashboard);
