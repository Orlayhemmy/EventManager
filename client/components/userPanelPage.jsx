import React from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';
import { Redirect, Link } from 'react-router-dom';

import { getEvents, eventSelected } from '../actions/eventActions';
import EventForm from '../components/eventPage/editEventForm';
import Navbar from './navbar.jsx';
import Footer from './footer.jsx';
import FlashMessageList from './flash/flashMessagesList';
import DeleteModal from './deleteModal';
import { centerSelected } from '../actions/centerActions';

@connect((store) => {
  return {
    auth: store.auth,
    events: store.event.events,
  };
})

export default class EventPage extends React.Component {

  componentWillMount() {
    this.props.dispatch(getEvents()); 
  }

  onClick(e) {
    let child = document.getElementById(e.target.id);
    let parent = child.parentNode;
    this.props.dispatch(eventSelected(e.target.id));
    this.getCenter(parent.id);
  }

  getCenter(id) {
    
    this.props.dispatch(centerSelected(id));
    console.log(id)
  }

  showHiddenDiv(e) {
    let id = e.target.dataset.toggleId;
    let id2 = e.target.id;
    if(!id) return;
    const div = document.getElementById(id);
    div.hidden = !div.hidden;
    if (id) {
      const div2 = document.getElementById(id2);
      if (!div.hidden) {
        return div2.style.display="none";
      }
      return div2.style.display="";
    } 
  }
  
   
  render() {
    if (!this.props.auth.isAuth) {
      <Redirect to="/" />
    }

    let eventId, editEventId, eventBody, form;
    const { pathname } = this.props.location;
    const content = _.map(this.props.events, (event) => {
      eventBody = `event-body${event.id}`;
      eventId = `eventDetails${event.id}`;
      editEventId = `editEventDetails${event.id}`;
      form = `form${event.id}`;
      let dateBooked = `date${event.id}`;
      return (
        <div>
          <div id={eventId} key={eventId}>
            
            <div class="form-outer text-center">
              <div class="form-inner">
                <div id={event.centerId}>
                  <img className="img" src="images/image2.jpg"/>
                  <h2>
                    <span className="media-heading" data-toggle-id={eventBody} onClick={this.showHiddenDiv}>
                      {event.eventTitle} 
                    </span>
                  </h2>
                  <Link to="/modify-event" id={event.centerId}><span onClick={this.onClick.bind(this)} id={event.id}>edit</span></Link>
                </div>
                <div id={eventBody} hidden>
                  <div className="media-body">
                      <h3><span>Date: </span> {event.bookedDate}</h3>
                      <h3><span>Center: </span> {event.Center.centerName}</h3>
                      <h3><span>Location: </span> {event.Center.location}</h3>
                      <h3><span>facilities: </span> {event.Center.facilities}</h3>
                      <h3><span>Event description: </span> {event.description}</h3>
                  </div>
                </div>
                <i id={eventId} data-toggle-id={editEventId} className="fa fa-pencil main-color edit" onClick={this.showHiddenDiv}></i>
                <span onClick={this.onClick.bind(this)} className="trash" data-toggle="modal" data-target="#deleteModal"><i id={event.id} className="fa fa-trash trash"></i></span>
              </div>
            </div>
           
          </div>
          <div id={editEventId} key={form} hidden>
            <div class="form-outer text-center">
              <div class="form-inner">
                <i className="fa fa-list-alt main-color"></i><br/>
                <span className="media-heading" data-toggle-id={eventBody} onClick={this.showHiddenDiv}>
                  {event.eventTitle} 
                </span>
                <EventForm id={event.id} title={event.eventTitle} description={event.description} isApproved={event.isApproved} date={event.bookedDate} center={event.Center.id}/> 
              </div>
              <i id={eventId} data-toggle-id={editEventId} className="fa fa-home main-color" onClick={this.showHiddenDiv}> home</i>
            </div>
          </div>
        </div>
      )
    });
    return (
        <div id="event-page">
          <Navbar />
          <div className="container">
            <div className="row">
              {content}
              <DeleteModal path={pathname}/>
            </div>
          </div>
          <Footer />
        </div>
    );
  }
}

