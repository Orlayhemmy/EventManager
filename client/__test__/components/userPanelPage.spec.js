import React from 'react';
import { Dashboard } from '../../components/Dashboard/Container/userPanelPage';

describe('Dashboard Component', () => {
  const props = {
    auth: {
      user: {
        isAdmin: true,
        id: 1
      },
      isAuth: true
    },
    userEvent: {
      status: '',
      message: ''
    },
    location: {
      pathname: '/'
    },
    state: {
      counter: 1,
      eventId: 1,
      eventName: 'Pizza Party'
    },
    activity: {},
    clearCenterStorage: jest.fn(() => {}),
    clearEventState: jest.fn(() => {}),
    getEvents: jest.fn(() => {}),
    getActivity: jest.fn(() => {}),
    eventBooked: jest.fn(() => {}),
    setCurrentEvent: jest.fn(() => {}),
    deleteEvent: jest.fn(() => {}),
  };
  const wrapper = shallow(<Dashboard {...props} />);

  it('should render component without error', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render modal on update', () => {
    wrapper.setProps({
      ...wrapper.props(),
      userEvent: {
        ...wrapper.props(),
        status: 200,
        message: 'Done'
      }
    });
    const modal = wrapper.find('DeleteModal');
    expect(modal.exists()).toBeTruthy();
  });
});
