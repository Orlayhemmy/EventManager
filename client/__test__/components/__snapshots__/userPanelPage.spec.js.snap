// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Component should render component without error 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dashboard
    activity={Object {}}
    auth={
      Object {
        "isAuth": true,
        "user": Object {
          "id": 1,
          "isAdmin": true,
        },
      }
    }
    clearCenterStorage={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    clearEventState={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    deleteEvent={[MockFunction]}
    eventBooked={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
          ],
        ],
      }
    }
    getActivity={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
          ],
        ],
      }
    }
    getEvents={
      [MockFunction] {
        "calls": Array [
          Array [
            0,
          ],
        ],
      }
    }
    location={
      Object {
        "pathname": "/",
      }
    }
    setCurrentEvent={[MockFunction]}
    state={
      Object {
        "counter": 1,
        "eventId": 1,
        "eventName": "Pizza Party",
      }
    }
    userEvent={
      Object {
        "message": "",
        "status": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(NavBar)
          path="/"
        />,
        <div
          className="container"
        >
          <div
            className="row pt-4"
          >
            <div
              className="col-lg-9"
            >
              <div
                className="row event"
              >
                <h3
                  className="main-color"
                >
                  List of events booked
                </h3>
                <hr />
              </div>
              <div
                className="row"
              >
                <Content
                  counter={0}
                  nextEvents={[Function]}
                  onClick={[Function]}
                  onSelect={[Function]}
                  userEvent={
                    Object {
                      "message": "",
                      "status": "",
                    }
                  }
                />
                <DeleteModal
                  content={
                    <Delete
                      onCancel={[Function]}
                      onDelete={[Function]}
                      title={undefined}
                    />
                  }
                />
              </div>
            </div>
            <div
              className="col-lg-3 col3-bg"
            >
              <div
                className="p-4 fw"
              >
                <h2>
                  Notifications
                </h2>
              </div>
              <Notifications
                activities={undefined}
                onClick={[Function]}
              />
            </div>
          </div>
        </div>,
        <Footer />,
      ],
      "id": "dashboard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row pt-4"
          >
            <div
              className="col-lg-9"
            >
              <div
                className="row event"
              >
                <h3
                  className="main-color"
                >
                  List of events booked
                </h3>
                <hr />
              </div>
              <div
                className="row"
              >
                <Content
                  counter={0}
                  nextEvents={[Function]}
                  onClick={[Function]}
                  onSelect={[Function]}
                  userEvent={
                    Object {
                      "message": "",
                      "status": "",
                    }
                  }
                />
                <DeleteModal
                  content={
                    <Delete
                      onCancel={[Function]}
                      onDelete={[Function]}
                      title={undefined}
                    />
                  }
                />
              </div>
            </div>
            <div
              className="col-lg-3 col3-bg"
            >
              <div
                className="p-4 fw"
              >
                <h2>
                  Notifications
                </h2>
              </div>
              <Notifications
                activities={undefined}
                onClick={[Function]}
              />
            </div>
          </div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-lg-9"
              >
                <div
                  className="row event"
                >
                  <h3
                    className="main-color"
                  >
                    List of events booked
                  </h3>
                  <hr />
                </div>
                <div
                  className="row"
                >
                  <Content
                    counter={0}
                    nextEvents={[Function]}
                    onClick={[Function]}
                    onSelect={[Function]}
                    userEvent={
                      Object {
                        "message": "",
                        "status": "",
                      }
                    }
                  />
                  <DeleteModal
                    content={
                      <Delete
                        onCancel={[Function]}
                        onDelete={[Function]}
                        title={undefined}
                      />
                    }
                  />
                </div>
              </div>,
              <div
                className="col-lg-3 col3-bg"
              >
                <div
                  className="p-4 fw"
                >
                  <h2>
                    Notifications
                  </h2>
                </div>
                <Notifications
                  activities={undefined}
                  onClick={[Function]}
                />
              </div>,
            ],
            "className": "row pt-4",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="row event"
                  >
                    <h3
                      className="main-color"
                    >
                      List of events booked
                    </h3>
                    <hr />
                  </div>,
                  <div
                    className="row"
                  >
                    <Content
                      counter={0}
                      nextEvents={[Function]}
                      onClick={[Function]}
                      onSelect={[Function]}
                      userEvent={
                        Object {
                          "message": "",
                          "status": "",
                        }
                      }
                    />
                    <DeleteModal
                      content={
                        <Delete
                          onCancel={[Function]}
                          onDelete={[Function]}
                          title={undefined}
                        />
                      }
                    />
                  </div>,
                ],
                "className": "col-lg-9",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h3
                        className="main-color"
                      >
                        List of events booked
                      </h3>,
                      <hr />,
                    ],
                    "className": "row event",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "List of events booked",
                        "className": "main-color",
                      },
                      "ref": null,
                      "rendered": "List of events booked",
                      "type": "h3",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "hr",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <Content
                        counter={0}
                        nextEvents={[Function]}
                        onClick={[Function]}
                        onSelect={[Function]}
                        userEvent={
                          Object {
                            "message": "",
                            "status": "",
                          }
                        }
                      />,
                      <DeleteModal
                        content={
                          <Delete
                            onCancel={[Function]}
                            onDelete={[Function]}
                            title={undefined}
                          />
                        }
                      />,
                    ],
                    "className": "row",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "counter": 0,
                        "nextEvents": [Function],
                        "onClick": [Function],
                        "onSelect": [Function],
                        "userEvent": Object {
                          "message": "",
                          "status": "",
                        },
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "content": <Delete
                          onCancel={[Function]}
                          onDelete={[Function]}
                          title={undefined}
                        />,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="p-4 fw"
                  >
                    <h2>
                      Notifications
                    </h2>
                  </div>,
                  <Notifications
                    activities={undefined}
                    onClick={[Function]}
                  />,
                ],
                "className": "col-lg-3 col3-bg",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <h2>
                      Notifications
                    </h2>,
                    "className": "p-4 fw",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Notifications",
                    },
                    "ref": null,
                    "rendered": "Notifications",
                    "type": "h2",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "activities": undefined,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(NavBar)
            path="/"
          />,
          <div
            className="container"
          >
            <div
              className="row pt-4"
            >
              <div
                className="col-lg-9"
              >
                <div
                  className="row event"
                >
                  <h3
                    className="main-color"
                  >
                    List of events booked
                  </h3>
                  <hr />
                </div>
                <div
                  className="row"
                >
                  <Content
                    counter={0}
                    nextEvents={[Function]}
                    onClick={[Function]}
                    onSelect={[Function]}
                    userEvent={
                      Object {
                        "message": "",
                        "status": "",
                      }
                    }
                  />
                  <DeleteModal
                    content={
                      <Delete
                        onCancel={[Function]}
                        onDelete={[Function]}
                        title={undefined}
                      />
                    }
                  />
                </div>
              </div>
              <div
                className="col-lg-3 col3-bg"
              >
                <div
                  className="p-4 fw"
                >
                  <h2>
                    Notifications
                  </h2>
                </div>
                <Notifications
                  activities={undefined}
                  onClick={[Function]}
                />
              </div>
            </div>
          </div>,
          <Footer />,
        ],
        "id": "dashboard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row pt-4"
            >
              <div
                className="col-lg-9"
              >
                <div
                  className="row event"
                >
                  <h3
                    className="main-color"
                  >
                    List of events booked
                  </h3>
                  <hr />
                </div>
                <div
                  className="row"
                >
                  <Content
                    counter={0}
                    nextEvents={[Function]}
                    onClick={[Function]}
                    onSelect={[Function]}
                    userEvent={
                      Object {
                        "message": "",
                        "status": "",
                      }
                    }
                  />
                  <DeleteModal
                    content={
                      <Delete
                        onCancel={[Function]}
                        onDelete={[Function]}
                        title={undefined}
                      />
                    }
                  />
                </div>
              </div>
              <div
                className="col-lg-3 col3-bg"
              >
                <div
                  className="p-4 fw"
                >
                  <h2>
                    Notifications
                  </h2>
                </div>
                <Notifications
                  activities={undefined}
                  onClick={[Function]}
                />
              </div>
            </div>,
            "className": "container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="col-lg-9"
                >
                  <div
                    className="row event"
                  >
                    <h3
                      className="main-color"
                    >
                      List of events booked
                    </h3>
                    <hr />
                  </div>
                  <div
                    className="row"
                  >
                    <Content
                      counter={0}
                      nextEvents={[Function]}
                      onClick={[Function]}
                      onSelect={[Function]}
                      userEvent={
                        Object {
                          "message": "",
                          "status": "",
                        }
                      }
                    />
                    <DeleteModal
                      content={
                        <Delete
                          onCancel={[Function]}
                          onDelete={[Function]}
                          title={undefined}
                        />
                      }
                    />
                  </div>
                </div>,
                <div
                  className="col-lg-3 col3-bg"
                >
                  <div
                    className="p-4 fw"
                  >
                    <h2>
                      Notifications
                    </h2>
                  </div>
                  <Notifications
                    activities={undefined}
                    onClick={[Function]}
                  />
                </div>,
              ],
              "className": "row pt-4",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="row event"
                    >
                      <h3
                        className="main-color"
                      >
                        List of events booked
                      </h3>
                      <hr />
                    </div>,
                    <div
                      className="row"
                    >
                      <Content
                        counter={0}
                        nextEvents={[Function]}
                        onClick={[Function]}
                        onSelect={[Function]}
                        userEvent={
                          Object {
                            "message": "",
                            "status": "",
                          }
                        }
                      />
                      <DeleteModal
                        content={
                          <Delete
                            onCancel={[Function]}
                            onDelete={[Function]}
                            title={undefined}
                          />
                        }
                      />
                    </div>,
                  ],
                  "className": "col-lg-9",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <h3
                          className="main-color"
                        >
                          List of events booked
                        </h3>,
                        <hr />,
                      ],
                      "className": "row event",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "List of events booked",
                          "className": "main-color",
                        },
                        "ref": null,
                        "rendered": "List of events booked",
                        "type": "h3",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": "hr",
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <Content
                          counter={0}
                          nextEvents={[Function]}
                          onClick={[Function]}
                          onSelect={[Function]}
                          userEvent={
                            Object {
                              "message": "",
                              "status": "",
                            }
                          }
                        />,
                        <DeleteModal
                          content={
                            <Delete
                              onCancel={[Function]}
                              onDelete={[Function]}
                              title={undefined}
                            />
                          }
                        />,
                      ],
                      "className": "row",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "counter": 0,
                          "nextEvents": [Function],
                          "onClick": [Function],
                          "onSelect": [Function],
                          "userEvent": Object {
                            "message": "",
                            "status": "",
                          },
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "content": <Delete
                            onCancel={[Function]}
                            onDelete={[Function]}
                            title={undefined}
                          />,
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                    ],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="p-4 fw"
                    >
                      <h2>
                        Notifications
                      </h2>
                    </div>,
                    <Notifications
                      activities={undefined}
                      onClick={[Function]}
                    />,
                  ],
                  "className": "col-lg-3 col3-bg",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <h2>
                        Notifications
                      </h2>,
                      "className": "p-4 fw",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Notifications",
                      },
                      "ref": null,
                      "rendered": "Notifications",
                      "type": "h2",
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "activities": undefined,
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
